include/ddc/kernels/fft.hpp

```diff
@@ -176,6 +176,18 @@ void rescale(
             });
 }
 
+template <class DDim>
+Real forward_full_norm_coef(DiscreteDomain<DDim> const& ddom) noexcept
```

Comment:
I may not fully understand the return type. How do you ensure that `Real` is identical to `real_type_t<Tout>`?

Reply:
So `Real` is the type that is used by discrete dimensions when they need to use floating points, for example the function coordinate uses that type. Should be a template parameter at some point, for now it is a global compile time parameter.

`real_type` is used to get the floating point type when passing an array to fft functions.

About the consistency, the user can mix them. I don't know how bad/good it is.

Reply:
Is there a way to derive a base floating point type from `DiscreteDomain<DDim>` and use it instead of `Real`. 
I expect that this function returns the coefficient in the same precision as `DiscreteDomain<DDim>`.
This would also make future changes easier

Reply:
I am not sure to see the advantage ? The precision of this function is tied to the precision of rlength which is Real.

The caller will cast if needed ? For now it is implicit, i can make it explicit to better identify it ?

Reply:
> I am not sure to see the advantage ? The precision of this function is tied to the precision of rlength which is Real.

Well, there is no advantage for the moment. Just to inform users that the precision of this function is tied to the precision of `rlength`. It is not very clear how to identify the precision of `rlength` is `Real`.

> The caller will cast if needed ? For now it is implicit, i can make it explicit to better identify it ?

I think it is better to cast explicitly if there is a change that `real_type_t<Tout>` is not identical to `Real`

Reply:
done in 4c95c14e7f2db103b08875b6f35fa91506e84c6f